### ISSUE TITLE
`<a-l>` on a line's newline selects the previous character

### ISSUE THREAD

**Issue by lenormf** [mawww/kakoune#699]
Hi,

On a non empty line, hitting `<a-l>` when the cursor is on the newline character selects the last character on that line.

HTH.

**Comment by GrantMoyer** [comment]
Relatedly, `gl` and `Gl` will also move the cursor to the last non-newline character on a line if it was previously on the newline.

**Comment by mawww** [comment]
While I agree its a little surprising, isnt that the most consistent behaviour we can come up with ? `<a-l>` and `gl` both go to the char before the end of line on the current line.

We could stay on the end of line if we already are on the end of line, but is that really more useful ?

**Comment by lenormf** [comment]
Since the `<a-l>` command is supposed to select everything from the current position to EOL, having a backward selection feels weird. I can see why goto commands would jump backward, but a leftward selection is expected to… go left.

I discovered this behavior by writing code that would check whether a comment line (in C) had text or not:
* select `^\h*//+\h*`
* `<a-l>`
* if the current selection is a newline character or nothing, there's no text

I was surprised when I found out that the last check always failed, because of `<a-l>`.
